server/util/networking/networking.go

```diff
@@ -410,10 +439,42 @@ func (a *HostNetAllocator) Get() (*HostNet, error) {
 	return nil, status.ResourceExhaustedError("host IP address space exhausted")
 }
 
+// tryFlock attempts to acquire the lockfile for the given net index.
+// It does not block - if the lock is already held then it returns a nil file
+// handle.
+// If a non-nil file handle is returned, the file must be closed to release the
+// lock.
+func (a *HostNetAllocator) tryFlock(netIdx int) (*os.File, error) {
+	if err := os.MkdirAll(*networkLockDir, 0755); err != nil {
+		return nil, fmt.Errorf("make lock dir %q: %s", *networkLockDir, err)
+	}
+	path := filepath.Join(*networkLockDir, fmt.Sprintf("ip_range.%d.lock", netIdx))
+	f, err := os.Create(path)
```

Comment:
Sanity check - if path already exists (because another process already holds the lock), will os.Create return the same file descriptor?

Reply:
it'll return a different file descriptor (file descriptors are local to a process) but the returned descriptor will point to the same underlying file (inode), so `flock` should work here. (I also checked this with some debug logging)